packages/fuels-signers/src/provider.rs

```diff
@@ -132,6 +141,48 @@ impl Provider {
         }
     }
 
+    /// Get the balance of all spendable coins `asset_id` for address `address`. This is different
+    /// from getting coins because we are just returning a number (the sum of UTXOs) instead of the
+    /// UTXOs.
+    pub async fn get_asset_balance(
+        &self,
+        address: &Address,
+        asset_id: AssetId,
+    ) -> Result<u64, ProviderError> {
+        Ok(self
+            .client
+            .balance(&*address.to_string(), Some(&*asset_id.to_string()))
+            .await?)
+    }
+
+    /// Get all the spendable balances of all assets for address `address`. This is different from
+    /// getting
+    /// the coins because we are only returning the sum of UTXOs and not the UTXOs themselves
+    pub async fn get_balances(
+        &self,
+        address: &Address,
+    ) -> Result<HashMap<String, u64>, ProviderError> {
+        // We don't paginate results because there are likely at most ~100 different assets in one
+        // wallet
+        let pagination = PaginationRequest {
```

Comment:
This is more of a question for my own sanity:

is this a temporary solution? or has this been discussed with @adlerjohn and/or @digorithm and we decided that this should be fine **forever** ?

Reply:
This has been discussed with @digorithm. We decided that this was fine until someone complains (and we don't think someone will, the pagination can alwas be done later).